sopel_remind/plugin.py

```diff
@@ -220,3 +220,25 @@ def remind_at(bot: SopelWrapper, trigger: Trigger):
         time=when,
     )
     bot.reply('I will remind you that at %s' % display_when)
+
+
+@plugin.command('reminders clear')
+def reminders_clear(sopel: SopelWrapper, trigger: Trigger) -> None:
+    """Clear all your reminders for the current channel.
+
+    You'll get a notice with the number of reminders cleared for that channel.
+    """
+    nick = trigger.nick
+    destination = trigger.sender
+
+    with LOCK:
+        removed = backend.clear_reminders(
+            sopel,
+            nick=nick,
+            destination=destination,
+        )
+
+    sopel.reply(
+        '%d reminder(s) cleared for this channel.' % removed,
+        notice=True,
+    )
```

Comment:
As we chatted about on IRC, probably do `bot.notice()` to `trigger.nick` directly instead of this, and include the destination as part of the feedback message.